Makefile

```diff
@@ -0,0 +1,120 @@
+# Makefile for devin and managing the Collimator project
+
+setup: setup-repo setup-deps setup-db
+
+setup-repo:
+	git pull
+	git submodule update --init --recursive
+	cd apps/notebook-runner && git submodule update --init --recursive
+
+setup-deps:
+	cd backend && yarn install
+	cd frontend && yarn install
+	cd apps/scratch && yarn install
+	cd apps/notebook-runner && make init
+
+setup-db:
+	cd backend && yarn prisma migrate dev
+
+build: build-frontend build-backend build-scratch build-jupyter
+
+build-frontend:
+	cd frontend && yarn build
+
+build-backend:
+	cd backend && yarn build
+
+build-scratch:
+	cd apps/scratch && yarn build
+
+build-jupyter:
+	cd apps/notebook-runner && make build
+
+build-e2e: build-e2e-frontend build-e2e-backend build-e2e-scratch
+
+build-e2e-frontend:
+	cd e2e && yarn build:frontend
+
+build-e2e-backend:
+	cd e2e && yarn build:backend
+
+build-e2e-scratch:
+	cd e2e && yarn build:scratch
+
+test: test-unit test-e2e
+
+test-unit: test-frontend test-backend test-scratch test-storybook
+
+test-e2e:
+	cd e2e && yarn test
+
+test-e2e-nobuild:
+	cd e2e && yarn test:nobuild
+
+test-frontend:
+	cd frontend && yarn test
+
+test-backend:
+	cd backend && yarn test
+
+test-scratch:
+	cd apps/scratch && yarn test
+
+test-storybook:
+	cd storybook && yarn test
+
+lint: lint-frontend lint-backend
+
+lint-frontend:
+	cd frontend && yarn lint
+
+lint-backend:
+	cd backend && yarn lint
+
+dev-backend:
+	cd backend && yarn start:dev
+
+dev-frontend:
+	cd frontend && yarn dev
+
+dev-scratch:
+	cd apps/scratch && yarn dev
+
+dev: dev-backend dev-frontend dev-scratch
+
+storybook:
+	cd storybook && yarn storybook
+
+jupyter-serve:
+	cd apps/notebook-runner && make serve
```

Comment:
Are these scripts we want to run at the project level? (Maybe they are, just want to start a discussion)